### PR TITLE
[13.0][FWD] shopinvader - Document generate/download (pdf)

### PR DESCRIPTION
**Pull request by i-vyshnevska**
forward port of #570 

Create an abstract component to generate and download a report from the front side.

- Extract the download part of invoice service;
- Create an abstract component;
- Apply it on invoice service;
- Apply it on sale service;
- Extract tests about download (and make a new class);
- Apply this test class on invoice + sale tests.

forward  https://github.com/shopinvader/odoo-shopinvader/pull/621

### CODE REVIEW

**Comment by rousseldenis** [comment]
@i-vyshnevska Thanks for this. Could you just cherry-pick the original commit to respect original author ?

**Comment by i-vyshnevska** [comment]
@rousseldenis I tried but had a lot of conflicts, for me it was easier to port like this, can we continue with this?

File: shopinvader_quotation/tests/test_quotation_download.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.shopinvader.tests.common import CommonCase, CommonTestDownload
+
+
+class QuotationDownloadCase(CommonCase, CommonTestDownload):
+    def setUp(self, *args, **kwargs):
+        super(QuotationDownloadCase, self).setUp(*args, **kwargs)
+        self.quotation = self.env.ref("shopinvader.sale_order_2")
```
Comment (simahawk) [inline shopinvader_quotation/tests/test_quotation_download.py]: move everything but service init to setUpClass

**Review by simahawk — CHANGES_REQUESTED** [review]
LG, small change required in test + squash pls :pray: 
While squashing pls include module name into commits (you can short `shopinvader_` to `s_`,  eg: `s_delivery_carrier: add pdf download`

File: shopinvader_quotation/tests/test_quotation_download.py (hunk 1 above)
Comment (i-vyshnevska) [reply shopinvader_quotation/tests/test_quotation_download.py]: @simahawk done

**Review by simahawk — APPROVED** [review]
LG

**Review by acsonefho — APPROVED** [review]
Code LGTM (not tested)

**Review by lmignon — APPROVED** [review]

**Pull request merged**